src/core/lib/security/authorization/rbac_translator.cc

```diff
@@ -0,0 +1,351 @@
+    if (!matcher_or.ok()) {
+      return absl::Status(matcher_or.status().code(),
+                          absl::StrCat("\"principals\" ", i, ": ",
+                                       matcher_or.status().message()));
+    }
+    principal_names.push_back(absl::make_unique<Rbac::Principal>(
+        Rbac::Principal::RuleType::PRINCIPAL_NAME,
+        std::move(matcher_or.value())));
+  }
+  return Rbac::Principal(Rbac::Principal::RuleType::OR,
+                         std::move(principal_names));
+}
+
+absl::StatusOr<Rbac::Principal> ParsePeer(const Json& json) {
+  std::vector<std::unique_ptr<Rbac::Principal>> peer;
+  auto it = json.object_value().find("principals");
+  if (it != json.object_value().end()) {
+    if (it->second.type() != Json::Type::ARRAY) {
+      return absl::InvalidArgumentError("\"principals\" is not an array.");
+    }
+    auto principal_names_or = ParsePrincipalsArray(it->second);
+    if (!principal_names_or.ok()) return principal_names_or.status();
+    if (!principal_names_or.value().principals.empty()) {
+      peer.push_back(absl::make_unique<Rbac::Principal>(
+          std::move(principal_names_or.value())));
+    }
+  }
+  if (peer.empty()) {
+    return Rbac::Principal(Rbac::Principal::RuleType::ANY);
+  }
+  return Rbac::Principal(Rbac::Principal::RuleType::AND, std::move(peer));
+}
+
+absl::StatusOr<Rbac::Permission> ParseHeaderValues(
+    const Json& json, absl::string_view header_name) {
+  if (json.array_value().empty()) {
+    return absl::InvalidArgumentError("\"values\" list is empty.");
+  }
+  std::vector<std::unique_ptr<Rbac::Permission>> values;
+  for (size_t i = 0; i < json.array_value().size(); ++i) {
+    const Json& child = json.array_value().at(i);
+    if (child.type() != Json::Type::STRING) {
+      return absl::InvalidArgumentError(
+          absl::StrCat("\"values\" ", i, ": is not a string."));
+    }
+    auto matcher_or = GetHeaderMatcher(header_name, child.string_value());
+    if (!matcher_or.ok()) {
+      return absl::Status(
+          matcher_or.status().code(),
+          absl::StrCat("\"values\" ", i, ": ", matcher_or.status().message()));
+    }
+    values.push_back(absl::make_unique<Rbac::Permission>(
+        Rbac::Permission::RuleType::HEADER, std::move(matcher_or.value())));
+  }
+  return Rbac::Permission(Rbac::Permission::RuleType::OR, std::move(values));
+}
+
+absl::StatusOr<Rbac::Permission> ParseHeaders(const Json& json) {
+  auto it = json.object_value().find("key");
+  if (it == json.object_value().end()) {
+    return absl::InvalidArgumentError("\"key\" is not present.");
+  }
+  if (it->second.type() != Json::Type::STRING) {
+    return absl::InvalidArgumentError("\"key\" is not a string.");
+  }
+  absl::string_view header_name = it->second.string_value();
+  // TODO(ashithasantosh): Add connection headers below.
+  if (absl::StartsWith(header_name, ":") ||
+      absl::StartsWith(header_name, "grpc-") || header_name == "host") {
```

Comment:
Please also remove "Host" header as well. 
Both "host" and "Host" are supported in HTTP 1.1